client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -0,0 +1,367 @@
+     * If the schema identified by the version is deleted using {@link SchemaRegistryClient#deleteSchemaVersion} api, 
+     * then if the encoding id was already generated for the pair of schema version and codec, then it will be returned. 
+     * However, if no encoding id for the versioninfo and codec pair was generated and the schema version was deleted, 
+     * then any call to getEncodingId using the deleted versionInfo will throw ResourceNotFoundException. 
+     * 
+     * @param groupId Id for the group. 
+     * @param versionInfo version of schema 
+     * @param codecType codec type
+     * @return Encoding id for the pair of version and codec type.
+     * @throws CodecTypeNotRegisteredException if codectype is not registered with the group. Use {@link SchemaRegistryClient#addCodecType} 
+     * @throws ResourceNotFoundException if group or version info is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    EncodingId getEncodingId(String groupId, VersionInfo versionInfo, String codecType) 
+            throws CodecTypeNotRegisteredException, ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Gets latest schema and version for the group (or type, if specified). 
+     * To get latest schema version for a specific type identified by {@link SchemaInfo#type}, provide the type. 
+     * Otherwise if the group is configured to allow multiple schemas {@link GroupProperties#allowMultipleTypes}, then 
+     * and type is not specified, then last schema added to the group across all types will be returned. 
+     * 
+     * @param groupId Id for the group. 
+     * @param schemaType Type of object identified by {@link SchemaInfo#type}. 
+     *                 
+     * @return Schema with version for the last schema that was added to the group (or type).
+     * @throws ResourceNotFoundException if group is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    SchemaWithVersion getLatestSchemaVersion(String groupId, @Nullable String schemaType)
+        throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Gets version corresponding to the schema.  
+     * For each schema type {@link SchemaInfo#type} and {@link SchemaInfo#serializationFormat} a versionInfo object uniquely 
+     * identifies each distinct {@link SchemaInfo#schemaData}. 
+     *
+     * @param groupId Id for the group. 
+     * @param schemaInfo SchemaInfo that describes format and structure. 
+     * @return VersionInfo corresponding to schema. 
+     * @throws ResourceNotFoundException if group is not found or if schema is not registered. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    VersionInfo getVersionForSchema(String groupId, SchemaInfo schemaInfo) throws ResourceNotFoundException, UnauthorizedException;
```

Comment:
I think this API should be removed.
It is not clear what use case it solves. Nor are its semantics clear: For example which fields will be used for an equality check on a schemaInfo. If we were to define which ones, it could cause a problem if new fields get added in the future.
What's more it is a very expensive way to obtain very simple information, uploading the whole of the schema just to find out what version it is doesn't really make sense. Surely it would be cheaper store an ID or (depending on the app) to just list all the schemas in the group. It also creates a very odd indexing problem for the service.

Reply:
finding equivalence of schemas is very important. 
it is required and crucial on the DataPath.
when a writer is started with a schema, it needs to get an id with which to tag the data.
All the writer has is the "schema". 
A writer is either authorized to add the schema or use one of existing schemas. 
but it needs to get the identifier for that schema from the service in either cases:
automatically register the schema OR get the version for the registered schema

As you are aware when you add a schema its version is returned. If schema was already registered previously, its existing version is returned. 
So equivalence check is required. 

And addSchema and getSchemaFromVersion perform the same equivalence check on following basic information of the schema - Type, Format + schemaBytes

To highlight the value of this API further -- this is an essential API that all other schema registries support too and requires users to upload the schema payload to get the version. 

Alternatively, client and server could agree upon a "fingerprinting" hash function and client could send the schema fingerprint and get a list of all schemas matching the fingerprint and perform local check to get the version corresponding to its schema.